### DISCUSSION TITLE
Unable to compile s3 to wasm

### DISCUSSION THREAD

**Discussion by Thintin914** [awslabs/aws-sdk-rust#1174] [Q&A] [answered]
I know sdk-aws-s3 uses tokio, which could not compile rust to wasm.
There has been discussion in [aws-sdk-rust github](https://github.com/awslabs/aws-sdk-rust/issues/59). However, I cannot use wasi because I'm developing a bevy app. It seems to be difficult to compile bevy into wasi.

I still encounter the **error[E0432]: unresolved import `crate::sys::IoSourceState`** when compiling to wasm-unknown-unknown.

Should I use aws s3 javascript and link it to rust with wasm-bindgen?
Or as this [reddit post said](https://www.reddit.com/r/rust/comments/1aia973/aws_sdk_s3_and_wasm/), I should use wasix?

**Comment by ysaito1001** [comment, marked as answer]
We have our test running in wasm that uses `aws-sdk-s3`. The contents of `Cargo.toml` of that test is programmatically generated [here](https://github.com/smithy-lang/smithy-rs/blob/main/tools/ci-cdk/canary-runner/src/build_bundle.rs#L158-L192).

Make sure to select only [allowed cargo features](https://docs.rs/tokio/latest/tokio/#wasm-support) of tokio and disable default features for `aws-sdk-s3` in your Cargo.toml via `default-features = false`
